modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/portlet/action/EditCommerceChannelCountryMVCActionCommand.java

```diff
@@ -0,0 +1,137 @@
+import com.liferay.commerce.product.service.CommerceChannelRelService;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Country;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.service.CountryService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextFactory;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.ParamUtil;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Stefano Motta
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + CPPortletKeys.COMMERCE_CHANNELS,
+		"mvc.command.name=/commerce_channels/edit_commerce_channel_country"
+	},
+	service = MVCActionCommand.class
+)
+public class EditCommerceChannelCountryMVCActionCommand
+	extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		String cmd = ParamUtil.getString(actionRequest, Constants.CMD);
+
+		try {
+			if (cmd.equals(Constants.DELETE)) {
+				_deleteCommerceChannelRel(
+					ParamUtil.getLong(actionRequest, "commerceChannelRelId"));
+			}
+			else if (cmd.equals(Constants.ADD_MULTIPLE)) {
+				_addCommerceChannelRels(
+					actionRequest,
+					ParamUtil.getLong(actionRequest, "commerceChannelId"),
+					ParamUtil.getLongValues(actionRequest, "countryIds"));
+			}
+		}
+		catch (Exception exception) {
+			if (exception instanceof NoSuchChannelException ||
+				exception instanceof PrincipalException) {
+
+				SessionErrors.add(actionRequest, exception.getClass());
+
+				actionResponse.setRenderParameter("mvcPath", "/error.jsp");
+			}
+			else {
+				throw exception;
+			}
+		}
+	}
+
+	private void _addCommerceChannelRels(
+			ActionRequest actionRequest, long commerceChannelId,
+			long[] countryIds)
+		throws Exception {
+
+		ServiceContext serviceContext = ServiceContextFactory.getInstance(
+			actionRequest);
+
+		for (long countryId : countryIds) {
+			try {
+				_commerceChannelRelService.addCommerceChannelRel(
+					Country.class.getName(), countryId, commerceChannelId,
+					serviceContext);
+
+				_countryService.updateGroupFilterEnabled(countryId, true);
+			}
+			catch (DuplicateCommerceChannelRelException
+						duplicateCommerceChannelRelException) {
+
+				if (_log.isDebugEnabled()) {
+					_log.debug(duplicateCommerceChannelRelException);
+				}
+			}
```

Comment:
I don't like this, why don't you move this for loop in the service method? plus can you try to avoid the duplicate exception? I thought that making the checkedIds disabled was to avoid this exception

Reply:
Yes, probably the exception can be removed because we are disabling the already checked countries. I'll give it a try

Reply:
I removed the try/catch block and created a new _commerceChannelRelService.addCommerceChannelRel**s** that accepts an array of classPK.
The loop here is still there but only for ```_countryService.updateGroupFilterEnabled(countryId, true);```

The advantage of have now "two" loops (one here and one inside the service) is that we are doing the permission check on the channel only once

Reply:
is it possible to make the same relation via API? if yes how would you enable the country group filter?

Reply:
No, I did some search but I think it's not possible to manage this relationship via API